platforms/android/tangram/src/main/java/com/mapzen/tangram/MapView.java

```diff
@@ -18,93 +18,35 @@
-    protected AsyncTask<Void, Void, Boolean> getMapTask;
 
     public MapView(Context context) {
-
         super(context);
-
         configureGLSurfaceView();
-
     }
 
     public MapView(Context context, AttributeSet attrs) {
-
         super(context, attrs);
-
         configureGLSurfaceView();
-
     }
 
     /**
-     * Interface for receiving a {@code MapController} once it is ready to be used
-     */
-    public interface OnMapReadyCallback {
-
-        /**
-         * Called when the map is ready to be used
-         * @param mapController A non-null {@code MapController} instance for this {@code MapView}
-         */
-        void onMapReady(MapController mapController);
-
-    }
-
-    /**
-     * Construct a {@code MapController} asynchronously; may only be called from the UI thread
-     * @param callback The object to receive the resulting MapController in a callback;
+     * Construct a {@code MapController}; may only be called from the UI thread
+     * @param listener The listener to receive to receive scene load events;
      * the callback will be made on the UI thread
-     * @param sceneFilePath Location of the YAML scene file within the asset bundle
      */
-    public void getMapAsync(@NonNull final OnMapReadyCallback callback,
-                            @NonNull final String sceneFilePath) {
-
-        getMapAsync(callback, sceneFilePath, null);
-    }
-
-    /**
-     * Construct a {@code MapController} asynchronously; may only be called from the UI thread
-     * @param callback The object to receive the resulting MapController in a callback;
-     * the callback will be made on the UI thread
-     * @param sceneFilePath Location of the YAML scene file within the asset bundle
-     * @param sceneUpdates List of SceneUpdate to be applied when loading this scene
-     */
-    public void getMapAsync(@NonNull final OnMapReadyCallback callback,
-                            @NonNull final String sceneFilePath,
-                            final List<SceneUpdate> sceneUpdates) {
-
-        disposeTask();
-
-        final MapController mapInstance = getMapInstance();
-
-        getMapTask = new AsyncTask<Void, Void, Boolean>() {
-
-            @Override
-            @SuppressWarnings("WrongThread")
-            protected Boolean doInBackground(Void... params) {
-                mapInstance.init();
-                mapInstance.loadSceneFile(sceneFilePath, sceneUpdates);
-                return true;
-            }
-
-            @Override
-            protected void onPostExecute(Boolean ok) {
-                addView(glSurfaceView);
-                disposeMap();
-                mapController = mapInstance;
-                callback.onMapReady(mapController);
-            }
-
-            @Override
-            protected void onCancelled(Boolean ok) {
-                mapInstance.dispose();
-            }
-
-        }.execute();
+    public MapController getMap(MapController.SceneLoadListener listener) {
```

Comment:
I'm not sure that it's bad - passing in `null` is acceptable so it's not limiting in any way - it just feels weird.